common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -1121,10 +1121,12 @@ public boolean matches(String expr) {
      */
     @Override
     public int hashCode() {
-        if (hash == 0) {
-            hash = PlatformDependent.hashCodeAscii(value, offset, length);
+        int h = hash;
```

Comment:
@fenik17 interesting read... that said I think it would only be problematic if we would have a constructor that directly set the `hash` or I am wrong ? That said I think it not harm so I am not against this change :)